packages/api/tests/lib/test-granule-delete.js

```diff
@@ -0,0 +1,503 @@
+  ]));
+});
+
+test.serial('deleteGranuleAndFiles() will not delete PG granule if the Dynamo granule delete fails', async (t) => {
+  const {
+    newPgGranule,
+    newDynamoGranule,
+    files,
+    s3Buckets,
+  } = await createGranuleAndFiles(
+    t.context.knex,
+    t.context.collectionCumulusId,
+    false
+  );
+
+  const mockGranuleDynamoModel = {
+    delete: () => {
+      throw new Error('Delete failed');
+    },
+  };
+
+  await t.throwsAsync(
+    deleteGranuleAndFiles({
+      knex: t.context.knex,
+      dynamoGranule: newDynamoGranule,
+      pgGranule: newPgGranule,
+      granuleModelClient: mockGranuleDynamoModel,
+    }),
+    { instanceOf: Error }
+  );
+
+  // granule should still exist in Dynamo and PG
+  t.true(await granulePgModel.exists(t.context.knex, { granule_id: newPgGranule.granule_id }));
+  t.true(await granuleModel.exists({ granuleId: newDynamoGranule.granuleId }));
+
+  // Files will still exist from S3 and PG.
+  await Promise.all(
+    files.map(async (file) => {
+      t.true(await s3ObjectExists({ Bucket: file.bucket, Key: file.key }));
+      t.true(await filePgModel.exists(t.context.knex, { bucket: file.bucket, key: file.key }));
+    })
+  );
+
+  t.teardown(() => deleteS3Buckets([
+    s3Buckets.protected.name,
+    s3Buckets.public.name,
+  ]));
+});
+
+test('deleteGranuleAndFiles() does not require a Postgres Granule', async (t) => {
+  // Create a granule in Dynamo only
+  const s3Buckets = {
+    protected: {
+      name: randomId('protected'),
+      type: 'protected',
+    },
+    public: {
+      name: randomId('public'),
+      type: 'public',
+    },
+  };
+  const granuleId = randomId('granule');
+  const files = [
+    {
+      bucket: s3Buckets.protected.name,
+      fileName: `${granuleId}.hdf`,
+      key: `${randomString(5)}/${granuleId}.hdf`,
+    },
+    {
+      bucket: s3Buckets.protected.name,
+      fileName: `${granuleId}.cmr.xml`,
+      key: `${randomString(5)}/${granuleId}.cmr.xml`,
+    },
+    {
+      bucket: s3Buckets.public.name,
+      fileName: `${granuleId}.jpg`,
+      key: `${randomString(5)}/${granuleId}.jpg`,
+    },
+  ];
+
+  const newGranule = fakeGranuleFactoryV2(
+    {
+      granuleId: granuleId,
+      status: 'failed',
+      published: false,
+      files: files,
+    }
+  );
+
+  await createS3Buckets([
+    s3Buckets.protected.name,
+    s3Buckets.public.name,
+  ]);
+
+  // Add files to S3
+  await Promise.all(newGranule.files.map((file) => s3PutObject({
+    Bucket: file.bucket,
+    Key: file.key,
+    Body: `test data ${randomString()}`,
+  })));
+
+  // create a new Dynamo granule
+  await granuleModel.create(newGranule);
+
+  await deleteGranuleAndFiles({
+    knex: t.context.knex,
+    dynamoGranule: newGranule,
+    pgGranule: undefined,
+  });
+
+  // Granule should have been removed from Dynamo
+  t.false(
+    await granuleModel.exists({ granuleId: granuleId })
+  );
+
+  // verify the files are deleted from S3.
+  await Promise.all(
```

Comment:
I'm not sure how this test is passing. From my reading of the code for `deleteGranuleAndFiles`, it is currently only running the deletion of S3 if `pgGranule` is **not** undefined.

Reply:
This _was_ deleting the S3 files in the Dynamo Granule model. The latest changes removed that step and made it an explicit call in granule-delete.js.

https://github.com/nasa/cumulus/pull/2057/commits/667e41a603d23b4fc2f553be40504132806f95d0

So far so good but I'll keep an eye out for test failures.